### PR TITLE
[180] Allow allocate edges between usages

### PR DESCRIPTION
**Pull request by jerome-obeo**
Bug: https://github.com/eclipse-syson/syson/issues/180

### CODE REVIEW

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * Used to find parent nodes inside a given Diagram.
+ *
+ * @author Jerome Gout
+ */
+public class ParentNodeFinder {
+
+    private final Diagram diagram;
+
+    public ParentNodeFinder(Diagram diagram) {
+        super();
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java:30]: super(); seems superfluous here, could you remove it please?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        this.diagram = diagram;
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java:31]: Could you add an Objects.requireNonNull here please?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractEdgeDescriptionProvider.java [hunk 3]
```diff
@@ -62,6 +63,15 @@ protected boolean isDeletable() {
         return true;
     }
 
+    /**
+     * Implementers can override this method to provide a specific label edit tool to apply on the edge.<br>
+     * By default no edit tool is provided.
+     * @return a label edit tool or {@code null} if no edit tool needed.
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractEdgeDescriptionProvider.java:69]: sometimes you use `{@code null}` and sometimes `<code>null</code>`. Is there a specific reason?
If not, could you juste use `<code>null</code>` everywhere please?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewCreateService.java [hunk 4]
```diff
@@ -285,4 +292,51 @@ public Element createPartUsageAsAllocationDefinitionEnd(AllocationDefinition sel
...
+            feature.getOwnedRelationship().add(reference);
+            reference.setReferencedFeature(usage);
+        }
+    }
+
+    /**
+     * Retrieve the parent node semantic element of the given node
+     * @param sourceNode a {@link Node}
+     * @param editingContext
+     * @return
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewCreateService.java:325]: you forgot to fill the @return

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/InterconnectionViewJavaServiceProvider.java [hunk 5]
```diff
@@ -44,7 +45,8 @@ public List<Class<?>> getServiceClasses(View view) {
                     InterconnectionViewEdgeService.class,
                     InterconnectionViewLabelService.class,
                     InterconnectionViewToolService.class,
-                    UtilService.class);
+                    UtilService.class,
+                    ViewEdgeService.class);
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/InterconnectionViewJavaServiceProvider.java:49]: May be `InterconnectionViewEdgeService` could inherits from `ViewEdgeService` instead. What do you think about?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/edges/AllocateEdgeDescriptionProvider.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    private final IDescriptionNameGenerator nameGenerator;
+
+    public AllocateEdgeDescriptionProvider(IColorProvider colorProvider, IDescriptionNameGenerator nameGenerator) {
+        super(colorProvider);
+        this.nameGenerator = Objects.requireNonNull(nameGenerator);
+    }
+
+    @Override
+    protected String getName() {
+        return new IVDescriptionNameGenerator().getEdgeName("Allocate");
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/edges/AllocateEdgeDescriptionProvider.java:43]: Why do you not use `this.nameGenerator.getEdgeName("Allocate")`?

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java (hunk 1 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java:30]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java (hunk 2 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ParentNodeFinder.java:31]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractEdgeDescriptionProvider.java (hunk 3 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractEdgeDescriptionProvider.java:69]: Done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewCreateService.java (hunk 4 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewCreateService.java:325]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/InterconnectionViewJavaServiceProvider.java (hunk 5 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/InterconnectionViewJavaServiceProvider.java:49]: Done

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/edges/AllocateEdgeDescriptionProvider.java (hunk 6 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/edges/AllocateEdgeDescriptionProvider.java:43]: Done

**Pull request merged**